engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -774,6 +774,17 @@ public TruffleLogger getLogger(Class<?> klass) {
     return TruffleLogger.getLogger(LanguageInfo.ID, klass);
   }
 
+  /**
+   * Gets a logger for the specified class that is bound to this engine.
+   * Such logger may then be safely used in threads defined in a thread-pool.
+   *
+   * @param clazz the class to name log entries with
+   * @return a new logger for the specified {@code path}
+   */
+  public TruffleLogger getLoggerBoundToEngine(Class<?> clazz) {
```

Comment:
Unrelated, but: Why not return `slf4j` logger from this method?

Reply:
I tried **not** to use `slf4j` within runtime at all.

Reply:
Using it as a simple facade over `TruffleLogger` would at least give us the same formatting of messages.